[Compression] Add new schema modifications (#1190)
timescale/docs: timescaledb/how-to-guides/compression/modify-a-schema.md

# Schema modifications
You can modify the schema of compressed hypertables in recent versions of
TimescaleDB.

> |Schema modification|Before TimescaleDB 2.1|TimescaleDB 2.1 to 2.5|TimescaleDB 2.6 and above|

Reviewer: Can we get non-breaking spaces between the product name and version number?

# Schema modifications
You can modify the schema of compressed hypertables in recent versions of
TimescaleDB.

|Schema modification|Before TimescaleDB&nbsp;2.1|TimescaleDB&nbsp;2.1 to 2.5|TimescaleDB&nbsp;2.6 and above|
|-|-|-|-|
|Add a nullable column|❌|✅|✅|
|Add a column with a default value and a `NOT NULL` constraint|❌|❌|✅|
|Rename a column|❌|✅|✅|
|Drop a column|❌|❌|✅|
|Change the data type of a column|❌|❌|❌|

To perform operations that aren't supported on compressed hypertables, first
[decompress][decompression] the table.

## Add a nullable column
To add a nullable column:
```sql
ALTER TABLE <hypertable> ADD COLUMN <column_name> <datatype>;
```

For example:
```sql
ALTER TABLE conditions ADD COLUMN device_id integer;
```
Note that adding constraints to the new column is not supported before
TimescaleDB 2.6.

> ## Add a column with a default value and a NOT NULL constraint

Author: I wish this worked! 😾 Markdown formatting doesn't work in headings right now, I'm going to put it on the list to look into because I really want to have inline code in headings

> ## Add a column with a default value and a NOT NULL constraint

Reviewer: Oh, bummer. I thought it did 😢

```sql
ALTER TABLE conditions RENAME device_id TO devid;
```

## Drop a column

> You can drop a column from a compressed hypertable, so long as the column is not

Reviewer: ```suggestion
You can drop a column from a compressed hypertable, if the column is not
```
